### PR TITLE
Delete possibly-conflicting entities in Db tests before test is run #4955

### LINKED ISSUE #4955: Delete possibly-conflicting entities in Db tests before test is run
For all Db Tests, entities are removed from the database after the test is run. This will be done after #4752, #4819, and #4954 is merged and all changes to the test-cases affected (and other tests in Db test) will be made in the same PR.

### PR DESCRIPTION
**Pull request by jkt001**
Fixes #4955 

I have moved the deletion of possibly conflicting entities before test is run in three files. I don't know whether the other files need them since there are no deletion of entities after each tests in each of those files.

### CODE REVIEW

File: src/test/java/teammates/test/cases/storage/StudentsDbTest.java [hunk 1]
```diff
@@ -63,8 +54,6 @@ public void testDefaultTimestamp() throws InvalidParametersException, EntityAlre
         ______TS("success : defaultTimeStamp for updatedAt date");
         
         assertEquals(defaultStudentCreationTimeStamp, student.getUpdatedAt());
-        
-        studentsDb.deleteStudent(s.course, s.email);
```
Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/storage/StudentsDbTest.java]: Not entirely sure, but why is this deleted? Isn't this PR about adding more of this stuff, or at least retaining the existing ones?

**Comment by thyageshm** [comment]
@jkt001 reminder for this pr

File: src/test/java/teammates/test/cases/storage/StudentsDbTest.java (hunk 1 above)
Comment (jkt001) [inline src/test/java/teammates/test/cases/storage/StudentsDbTest.java]: It <strong>is</strong> the purpose of this PR. Previously, after each tests, the corresponding entities are kept. We now want them to be retained, hence this is deleted.

Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/storage/StudentsDbTest.java]: Got it.

**Comment by wkurniawan07** [comment]
As long as it can pass the test reliably in one run, it's good to go.

**Comment by thyageshm** [comment]
@jkt001 please merge this in soon. coordinate with @wkurniawan07 as he will be merging #5139

**Comment by thyageshm** [comment]
I guess we can merge this in the next milestone

**Comment by thyageshm** [comment]
@jkt001 please merge this in

**Comment by damithc** [comment]
Closed and opened to force a Travis build. Don't merge unless travis passes.
